build-system/pr-check/build-targets.js

```diff
@@ -311,10 +312,25 @@ function determineBuildTargets() {
     return buildTargets;
   }
   buildTargets = new Set();
+
   lintFiles = globby.sync(config.lintGlobs);
   presubmitFiles = globby.sync(config.presubmitGlobs);
   prettifyFiles = globby.sync(config.prettifyGlobs);
   const filesChanged = gitDiffNameOnlyMain();
+
+  // To assist with debugging CircleCI jobs, add a file called
+  // .FORCE_BUILD_TARGETS with the target names that you want to force (See
+  // Targets class above) to the root directory, one per line.
+  if (filesChanged.indexOf('.FORCE_BUILD_TARGETS') >= 0) {
+    fs.readFileSync('.FORCE_BUILD_TARGETS', 'utf-8')
+      .trim()
+      .split('\n')
+      .forEach((target) => {
+        buildTargets.add(target);
+      });
+    return buildTargets;
+  }
```

Comment:
I understand the occasional need for debugging of this sort, but I do not think this kind of code should be checked in to the repo.

There's also the added risk of someone inadvertently checking in `.FORCE_BUILD_TARGETS`, which will then permanently override the detected build targets not just for the PR that adds it, but for all subsequent PRs.

An alternative solution to this is that when you want to trigger a particular target while debugging CI, simply make a whitespace-only change to a file that belongs to that target. E.g. To trigger `amp ava`, add a blank line to `ava.js`, etc. This will have the desired effect without the risk mentioned above.

Reply:
> There's also the added risk of someone inadvertently checking in `.FORCE_BUILD_TARGETS`, which will then permanently override the detected build targets not just for the PR that adds it, but for all subsequent PRs.

I thought you might say that. One thing we can do is to add a step to the teardown that fails the workflow eventually, if it has this file. That's a bit more of an involved change, but I'm willing to look into it if you think that's a direction we can take

> An alternative solution to this is that when you want to trigger a particular target while debugging CI, simply make a whitespace-only change to a file that belongs to that target. E.g. To trigger `amp ava`, add a blank line to `ava.js`, etc. This will have the desired effect without the risk mentioned above.

My problem here is that making a whitespace-only change can trigger too much, if I want to quickly iterate on a specific part of the CI system

Reply:
> I thought you might say that. One thing we can do is to add a step to the teardown that fails the workflow eventually, if it has this file. That's a bit more of an involved change, but I'm willing to look into it if you think that's a direction we can take

Your instinct is right, it's even more involved, and I don't think we should go in this direction.

> My problem here is that making a whitespace-only change can trigger too much, if I want to quickly iterate on a specific part of the CI system

Here is yet another alternative that I routinely use while debugging. Simply add a hard-coded early-return path to [`determineBuildTargets()`](https://github.com/ampproject/amphtml/blob/69c41e1fe1f4eafffb53c0970c3a1b0b71e3add3/build-system/pr-check/build-targets.js#L309) for as long as you are debugging:

```js
function determineBuildTargets() {
  if (buildTargets != undefined) {
    return buildTargets;
  }
  buildTargets = new Set();

  // DEBUGGING: DO NOT SUBMIT
  buildTargets.add(Targets.FOO).add(Targets.BAR);
  return buildTargets;

  // Everything below this is skipped while you're debugging
```

Reply:
Alright, I'll abandon this effort :)